elements/lisk-transactions/test/12_multisignature_transaction.spec.ts

```diff
@@ -197,353 +244,500 @@ describe('Multisignature transaction class', () => {
-			});
 
-			const { status, errors } = await transaction.processMultisignatures(
-				store,
-			);
-			expect(status).toBe(Status.FAIL);
+			const transaction = new MultisignatureTransaction(invalidTransaction);
+			const errors = (transaction as any).validateAsset();
+
 			expect(errors).toHaveLength(1);
-			expect(errors[0].dataPath).toBe('.signatures');
+			expect(errors[0].message).toBe(
+				`'.optionalKeys' should NOT have duplicate items (items ## 1 and 0 are identical)`,
+			);
 		});
 	});
 
-	describe('#applyAsset', () => {
-		it('should call state store', async () => {
-			await (validTestTransaction as any).applyAsset(store);
-			expect(store.account.get).toHaveBeenCalledWith(
-				validTestTransaction.senderId,
-			);
-			expect(store.account.set).toHaveBeenCalledWith(
-				multisignatureSender.address,
-				multisignatureSender,
+	describe('#validateAsset', () => {
+		it('should return errors when numberOfSignatures is bigger than the sum of all keys', async () => {
+			const invalidTransaction = {
+				...validMultisignatureRegistrationTransaction,
+				asset: {
+					...validMultisignatureRegistrationTransaction.asset,
+					numberOfSignatures: 5,
+				},
+			};
+			const transaction = new MultisignatureTransaction(invalidTransaction);
+			const errors = (transaction as any).validateAsset();
+
+			expect(errors).toHaveLength(1);
+			expect(errors[0].message).toBe(
+				`The numberOfSignatures is bigger than the count of Mandatory and Optional keys`,
 			);
 		});
 
-		it.skip('should return no errors', async () => {
-			const errors = await (validTestTransaction as any).applyAsset(store);
-
-			expect(errors).toHaveLength(0);
-		});
+		it('should return errors when numberOfSignatures is smaller than mandatory key count', async () => {
+			const invalidTransaction = {
+				...validMultisignatureRegistrationTransaction,
+				asset: {
+					...validMultisignatureRegistrationTransaction.asset,
+					numberOfSignatures: 1,
+				},
+			};
+			const transaction = new MultisignatureTransaction(invalidTransaction);
+			const errors = (transaction as any).validateAsset();
 
-		it('should return error when account is already multisignature', async () => {
-			// storeAccountGetStub.mockReturnValue(multisignatureSender);
-			const errors = await (validTestTransaction as any).applyAsset(store);
 			expect(errors).toHaveLength(1);
-			expect(errors[0].dataPath).toBe('.signatures');
+			expect(errors[0].message).toBe(
+				`The numberOfSignatures needs to be equal or bigger than the number of Mandatory keys`,
+			);
 		});
 
-		it('should return error when keysgroup includes sender key', async () => {
-			const invalidSender = {
-				...defaultAccount,
-				address: '',
-				...multisignatureSender,
-				membersPublicKeys: [
-					...(multisignatureSender as any).membersPublicKeys,
-					multisignatureSender.publicKey,
-				],
+		it('should return errors when mandatory and optional key sets are not disjointed', async () => {
```

Comment:
Prefer to add a case where
- mandatory and optional are both empty
- mandatory + optional is greater than 64

Reply:
Added